testing/citest/tests/kube_v2_smoke_test.py

```diff
@@ -0,0 +1,267 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Integration test to see if the image promotion process is working for the
+Spinnaker Kubernetes V2 integration.
+"""
+
+# Standard python modules.
+import sys
+
+# citest modules.
+import citest.kube_testing as kube
+import citest.json_contract as jc
+import citest.json_predicate as jp
+import citest.service_testing as st
+
+# Spinnaker modules.
+import spinnaker_testing as sk
+import spinnaker_testing.gate as gate
+import spinnaker_testing.frigga as frigga
+import citest.base
+
+
+class KubeV2SmokeTestScenario(sk.SpinnakerTestScenario):
+  """Defines the scenario for the kube v2 smoke test.
+  """
+
+  @classmethod
+  def new_agent(cls, bindings):
+    """Implements citest.service_testing.AgentTestScenario.new_agent."""
+    agent = gate.new_agent(bindings)
+    agent.default_max_wait_secs = 180
+    return agent
+
+  @classmethod
+  def initArgumentParser(cls, parser, defaults=None):
+    """Initialize command line argument parser.
+
+    Args:
+      parser: argparse.ArgumentParser
+    """
+    super(KubeV2SmokeTestScenario, cls).initArgumentParser(
+        parser, defaults=defaults)
+
+    defaults = defaults or {}
+    parser.add_argument(
+      '--test_namespace', default='default',
+      help='The namespace to manage within the tests.')
+
+  def __init__(self, bindings, agent=None):
+    """Constructor.
+
+    Args:
+      bindings: [dict] The data bindings to use to configure the scenario.
+      agent: [GateAgent] The agent for invoking the test operations on Gate.
+    """
+    super(KubeV2SmokeTestScenario, self).__init__(bindings, agent)
+    bindings = self.bindings
+
+    # We'll call out the app name because it is widely used
+    # because it scopes the context of our activities.
+    # pylint: disable=invalid-name
+    self.TEST_APP = bindings['TEST_APP']
+
+    # Take just the first if there are multiple
+    # because some uses below assume just one.
+    self.TEST_NAMESPACE = bindings['TEST_NAMESPACE'].split(',')[0]
+    self.pipeline_id = None
+
+    # We will deploy two images. One with a tag that we want to find,
+    # and another that we don't want to find.
+    self.__image_registry = 'gcr.io'
```

Comment:
I dont think these should be hardcoded. Hardcoding parameter defaults is ok.
One should be able to run this test from custom build configurations.

Reply:
This has nothing to do with a build environment, we are just picking an arbitrary image public image to deploy -- the actual choice has no bearing on the test